components/CompletedCheck.tsx

```diff
@@ -0,0 +1,27 @@
+import { Checkbox, CheckedState } from "tamagui";
+import { useContext } from "react";
+import { observer } from "mobx-react-lite";
+import { Check } from "@tamagui/lucide-icons";
+
+import { Todo, TodoContext } from "../stores/Todo.store";
+
+export const CompletedCheck = observer(function ({ todo }: { todo: Todo }) {
+  const store = useContext(TodoContext)!;
+  const onCheckedChange = (checked: CheckedState) => {
+    store.editChecked(todo, checked === true);
```

Comment:
would 
```suggestion
    store.editChecked(todo, checked);
```
do the job?

Reply:
No, since `CheckedState` can be `indeterminate`